projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1624,19 +1383,64 @@
 				settings.startIndex = 0; // -1 returned if can't find index, so start from beginning
 			}
 
-			domUtil.emitEvent( carousel.container, 'jp_carousel.beforeOpen' );
+			domUtil.emitEvent( carousel.overlay, 'jp_carousel.beforeOpen' );
+			carousel.gallery.innerHTML = '';
 
-			domUtil.fadeIn( carousel.container, function () {
-				domUtil.emitEvent( carousel.container, 'jp_carousel.afterOpen' );
-			} );
+			// Need to set the overlay manually to block or swiper does't initialise properly.
+			carousel.overlay.style.opacity = 1;
+			carousel.overlay.style.display = 'block';
 
-			carousel.gallery.innerHTML = '';
 			initCarouselSlides( gallery.querySelectorAll( settings.imgSelector ), settings.startIndex );
-			selectSlideAtIndex( settings.startIndex );
+
+			swiper = new window.Swiper( '.swiper-container', {
+				centeredSlides: true,
+				zoom: true,
+				loop: carousel.slides.length > 1 ? true : false,
+				pagination: {
+					el: '.swiper-pagination',
+					clickable: true,
+				},
+				navigation: {
+					nextEl: '.swiper-button-next',
+					prevEl: '.swiper-button-prev',
+				},
+				initialSlide: settings.startIndex,
+				on: {
+					init: function () {
+						selectSlideAtIndex( settings.startIndex );
+					},
+				},
+			} );
+
+			swiper.on( 'slideChange', function () {
+				var index;
+				// Swiper indexes slides from 1, plus when looping to left last slide ends up
+				// as 0 and looping to right first slide as total slides + 1. These are adjusted
+				// here to match index of carousel.slides.
+				if ( swiper.activeIndex === 0 ) {
+					index = carousel.slides.length - 1;
+				} else if ( swiper.activeIndex === carousel.slides.length + 1 ) {
+					index = 0;
+				} else {
+					index = swiper.activeIndex - 1;
+				}
+				selectSlideAtIndex( index );
+			} );
+
+			domUtil.fadeIn( carousel.overlay, function () {
+				domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
+			} );
 		}
 
 		// Register the event listener for starting the gallery
 		document.body.addEventListener( 'click', function ( e ) {
+			var isIE11 = window.navigator.userAgent.match( /Trident\/7\./ );
```

Comment:
It's generally discouraged to use user agent parsing to identify individual browsers, instead of performing feature detection for the required features.

Since IE 11 doesn't support CSS grid, and since the new carousel requires it to work correctly, we can achieve a similar result by checking for feature support for grid, with the bonus that this will also catch other, older browsers, or any IE 11 instances masquerading as something else. Here's the code I'd use (note that I renamed the variable):

```suggestion
			var isCompatible = window.CSS && window.CSS.supports && window.CSS.supports( 'display', 'grid' );
```

Reply:
Addressed in https://github.com/Automattic/jetpack/pull/20219

Reply:
Thanks, @apeatling!